eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/Retry.java

```diff
@@ -0,0 +1,118 @@
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.exception.ErrorCondition;
+import com.azure.eventhubs.implementation.ClientConstants;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class Retry {
+
+    public static final Retry NO_RETRY = new RetryExponential(Duration.ofSeconds(0), Duration.ofSeconds(0), 0);
+
+    private AtomicInteger retryCount = new AtomicInteger(0);
+
+    /**
+     * Check if the existing exception is a retryable exception.
+     *
+     * @param exception A exception that was observed for the operation to be retried.
+     * @return true if the exception is a retryable exception, otherwise false.
+     * @throws IllegalArgumentException when the exception is null.
+     */
+    public static boolean isRetryableException(Exception exception) {
+        if (exception == null) {
+            throw new IllegalArgumentException("exception cannot be null");
+        }
+
+        if (exception instanceof AmqpException) {
+            return ((AmqpException) exception).isTransient();
+        }
+        return false;
+    }
+
+    /**
+     * Get 'NO_RETRY' of current.
+     *
+     * @return Retry 'NO_RETRY'.
+     */
+    public static Retry getNoRetry() {
+        return Retry.NO_RETRY;
+    }
+
+    /**
+     * Get default configured Retry.
+     *
+     * @return Retry which has all default property set up.
+     */
+    public static Retry getDefault() {
+        return new RetryExponential(
+            ClientConstants.DEFAULT_RETRY_MIN_BACKOFF,
+            ClientConstants.DEFAULT_RETRY_MAX_BACKOFF,
+            ClientConstants.DEFAULT_MAX_RETRY_COUNT);
+    }
+
+    /**
+     * Increase one count to current count value.
+     *
+     * @return current AtomicInteger value.
+     */
+    public int incrementRetryCount() {
+        return retryCount.incrementAndGet();
+    }
+
+    /**
+     * Get the current retried count.
+     *
+     * @return current AtomicInteger value.
+     */
+    public int getRetryCount() {
+        return retryCount.get();
+    }
+
+    /**
+     * reset AtomicInteger to value zero.
+     */
+    public void resetRetryInterval() {
+        retryCount.set(0);
+    }
+
+    /**
+     * Calculates the amount of time to delay before the next retry attempt
+     *
+     * @param lastException The last exception that was observed for the operation to be retried.
+     * @param remainingTime The amount of time remaining for the cumulative timeout across retry attempts.
+     * @return The amount of time to delay before retrying the associated operation; if {@code null}, then the operation is no longer eligible to be retried.
+     */
+    public Duration getNextRetryInterval(Exception lastException, Duration remainingTime) {
+        int baseWaitTime = 0;
+
+        if (lastException == null || !(lastException instanceof AmqpException)) {
```

Comment:
If the lastException is null or not an instance of AmqpException, we should return `null`. For example, an IllegalArgumentException is not retriable.

Reply:
You should also test that the exception is retryable or not, and if they have hit the maximum number of attempts. These are checks that are agnostic of the retry algorithm.

Reply:
both tests are tested in test cases: 
(1) isRetriable: https://github.com/Azure/azure-sdk-for-java/pull/3818/files#diff-88f19a64404192ec0b032ec6f648b628R57
(2) Excess max retry:  https://github.com/Azure/azure-sdk-for-java/pull/3818/files#diff-88f19a64404192ec0b032ec6f648b628R72

Reply:
What I mean is that the logic for if the exception is retryable or not and max attempts (including maxRetryCount) should be in Retry.java rather than ExponentialRetry.java because the max retries and is transient applies to _all_ retry policies.

Reply:
fixed